src/js/modules/infragistics.ui.editors.js

```diff
@@ -12745,6 +12747,12 @@
 				this._super(this._parseDateFromMaskedValue(currentVal));
 			}
 		},
+		_valueIndexInList: function (val) { //igTimePicker
+			if (!val && val !== 0) {
+				return -1;
+			}
+			return $.inArray(val.toTimeString(), this.options.timeItems);

```

Comment:
Instead of creating second array, timeItems, you can check like this:
return $.inArray(val.toTimeString(), this.options.listItems.toTimeString());
or create a function to compare time values

Reply:
My implementation is better as it caches the items. `_validateValue` is invoked a lot of times and it doesn't make sense to convert list times to strings each time the value needs to be validated. The cache is updated each time the drop down items are recreated.
`$.inArray(val.toTimeString(), this.options.listItems.toTimeString());` is not going to work as `this.options.listItems.toTimeString()` won't return an array.
Creating a function to compare time portions is an overkill as we only need to compare them for equality and not for less or greater values.